### PR TITLE
Add second workaround for install package

### LINKED ISSUE #205: Installation manual
Hello,

Currently you suggest to use `--ignore-platform-reqs` command line switch as workaround for installation issues. What do you think about adding emulation of `ext-mongo` in composer.json file in `config.platform` (https://getcomposer.org/doc/06-config.md#platform) section? This workaround disable only check for particular modules instead of completely disable platform checks.

### PR DESCRIPTION
**Pull request by vkhramtsov**
This PR fixes #205

### CODE REVIEW

**Comment by vkhramtsov** [comment]
@alcaeus, Please review

File: README.md [hunk 1]
```diff
@@ -33,9 +33,23 @@ root:
...
 
-To fix this, you can use the `--ignore-platform-reqs` switch when running the
-above command, or when running `composer update` with no `composer.lock` file
-present.
+To fix this, you can use next ways:
+ - The `--ignore-platform-reqs` switch when running the above command, or
+ when running `composer update` with no `composer.lock` file present. Please
+ keep in mind that this switch completely disables platform requirements check.
+ - Update/create [config.platform section](https://getcomposer.org/doc/06-config.md#platform)
+ in your composer.json file with, for example `"ext-mongo": "1.8.0"`. This setting disables check
```
Comment (alcaeus) [inline README.md]: Should be `1.6.14`. See https://github.com/alcaeus/mongo-php-adapter/blob/9f42213a51e12c0fc1ce4773840c34d8ab3afd6f/composer.json#L21

File: README.md [hunk 2]
```diff
@@ -33,9 +33,23 @@ root:
...
+ only for old mongo extension. For example:
+
+```javascript
+    ...
+    "config": {
+        "platform": {
+            "ext-mongo": "1.8.0"
```
Comment (alcaeus) [inline README.md]: Same as above.

File: README.md (hunk 1 above)
Comment (vkhramtsov) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (vkhramtsov) [reply README.md]: Fixed

**Comment by vkhramtsov** [comment]
Waiting for review

**Comment by alcaeus** [comment]
> Waiting for review

No need to keep repeating. I'll test and review later today when I have time. Thanks.

**Comment by dzuelke** [comment]
We ignore `config.platform`, @alcaeus. I am currently rewriting `provide` to `replace` for this package only, which is a nasty nasty hack, but the only way unfortunately :( See https://github.com/heroku/heroku-buildpack-php/commit/7dfadaa3221c20cecea0bb18e39a192d06fd6c9a

**Comment by alcaeus** [comment]
@vkhramtsov closing this, I realized it makes more sense to provide this as `composer config` command in the installation instructions instead of providing it as a workaround. Thank you very much for the suggestion, this indeed works lots better than the previous solutions.

@dzuelke I don't remember what our tests with `provide` vs. `replace` a while back resulted in. I don't think composer will mind if the composer.json file has both. I can definitely add it if it helps you in removing that workaround you have there.

**Comment by dzuelke** [comment]
Isn't the problem with replace that the install will then fail if the legacy extension is present? And you don't want that, since it'll quietly do nothing if the extension is there?

**Comment by alcaeus** [comment]
> Isn't the problem with replace that the install will then fail if the legacy extension is present

Now that you mention it, yes. I believe that was it.

**Comment by dzuelke** [comment]
I had another longer chat with @naderman about this a few weeks back. It's more or less unfixable in its current incarnation. The problem now tracked in https://github.com/composer/composer/issues/6753 as well.

**Pull request closed without merging**